src/licenses.ts

```diff
@@ -0,0 +1,39 @@
+import {Change, ChangeSchema} from './schemas'
+
+/**
+ * Loops through a list of changes, filtering and returning the
+ * ones that don't conform to the licenses allow/deny lists.
+ * @param {Change[]} changes The list of changes to filter.
+ * @param { { allow?: string[], deny?: string[]}} licenses An object with `allow`/`deny` keys, each containing a list of licenses.
+ * @returns {Array<Change} The list of denied changes.
+ */
+export function getDeniedLicenseChanges(
+  changes: Array<Change>,
+  licenses: {
+    allow?: Array<string>
+    deny?: Array<string>
+  }
+): Array<Change> {
+  let {allow = null, deny = null} = licenses
+
+  let disallowed: Change[] = []
+
+  for (const change of changes) {
+    let license = change.license
+    // TODO: be loud about unknown licenses
+    if (license === null) {
+      continue
+    }
+    if (allow !== null) {
+      if (!allow.includes(license)) {
+        disallowed.push(change)
+      }
+    } else if (deny !== null) {
```

Comment:
Following the logic, this means that the allow list always super-cedes deny lists. That is, if there's any allow list, the deny list won't be used (the `if-else` ensures that the else branch will not be taken if allow !== null).

Is this the intended behavior? Fine if it is, but let's be sure to make that behavior explicit in documentation and tests.

Reply:
This is on purpose, from the PR description: 

> These two options are mutually exclusive, users need to pick the one that suits them better. An error will be raised if both options are specified in the configuration file.

Zod takes care of [doing the actual validation](https://github.com/actions/dependency-review-action/blob/adding-lists/src/schemas.ts#L40-L43). [Here's a test](https://github.com/actions/dependency-review-action/blob/adding-lists/__tests__/config.test.ts#L27-L31) that covers this use case from a higher level. 

I've added more information in the function's comment in 6b5518a.

Reply:
Perfect, thanks for clarifying 👍